server/events.go

```diff
@@ -248,6 +248,22 @@ type pubMsg struct {
 	last bool
 }
 
+var pubMsgPool sync.Pool
+
+func newPubMsg(c *client, sub, rply string, si *ServerInfo, hdr map[string]string,
+	msg interface{}, oct compressionType, echo, last bool) *pubMsg {
+
+	var m *pubMsg
+	pm := pubMsgPool.Get()
+	if pm != nil {
+		m = pm.(*pubMsg)
+	} else {
+		m = &pubMsg{}
+	}
+	(*m) = pubMsg{c, sub, rply, si, hdr, msg, oct, echo, last}
```

Comment:
Interesting approach with the temp object. Cool.

Reply:
I should have also put a comment that I do this way because setting all fields of an object when using pool is CRITICAL, and doing this way we would get a compiler error if a field is added to the pubMsg struct without updating this function.